### DISCUSSION TITLE
To configure correct planar CT imaging geometry

### DISCUSSION THREAD

**Discussion by pingchanglee** [CERN/TIGRE#534] [Q&A]
I try to reproduce CT reconstruction in the paper _Laminography in the Lab: Imaging planar objects using a conventional x-ray CT scanner_
[download link](https://www.researchgate.net/publication/330273195_Laminography_in_the_Lab_Imaging_planar_objects_using_a_conventional_x-ray_CT_scanner)
In this study, the object is tilted and the object stage is rotate along the tilted axis. 

Based the TIGRE demo example,  DEMO 18: Arbitrary axis of rotation, I tried to reconstruct the LEGO volume.
According to the demo example, it says: 

>  In TIGRE this has been implemented by defining the rotation with 3 angles, specifically the ZYZ configuration of Euler angles.

So, I first converted the rotation angles to Euler angles.

```Python
  ### Convert to ZYZ Euler angle 
  rad = np.pi / 180 
  tilt_radian = rad * 30
  unit_vec = np.array((np.sin(tilt_radian) , 0, np.cos(tilt_radian)))
  geo.angles = np.zeros((n_proj, 3))
  
  for i in range(n_proj):
      if i == 0:
          geo.angles[i,:] = np.array((angles[i], tilt_radian, 0))
      else:
          rot = R.from_rotvec( angles[i] * unit_vec)
          euler_angle = rot.as_euler('ZYZ')
          geo.angles[i,:] = euler_angle
```

But the reconstruction result is totally wrong.

![image](https://github.com/CERN/TIGRE/assets/122041758/ae27db2a-7cbe-4175-8b5f-4bec7eee6848)

However, when I set the angles this way:
```Python
  n_proj = angles.shape[0]
  zeros_array = np.zeros((n_proj, 1), dtype=np.float32)
  tilt_array = np.ones((n_proj, 1), dtype=np.float32) * 30 * np.pi / 180
  geo.angles = np.hstack((angles.reshape(n_proj, 1) , tilt_array, zeros_array))
```
the reconstruction result seems right!
This result confused me, since this is not ZYZ.

![image](https://github.com/CERN/TIGRE/assets/122041758/96a78a50-3e0f-4a44-9a60-2cefd16303c9)

Hopefully, there is someone can help me to figure this out. Thanks!
 
In all references listed in [Further Reading](https://github.com/CERN/TIGRE/blob/master/Frontispiece/Further_reading.md) and [code](https://github.com/CERN/TIGRE/blob/master/Python/tigre/utilities/common_geometry.py), talk very little about the spherical trajectory of both detector and X-ray source.  Is it possible to elaborate this part in TIGRE?
Thanks!!!

Both test code and the projection data are available [here](https://1drv.ms/u/s!AoPNNWDbBTsVhK4l5I6QWQa1iGtlww?e=GxS1Og)

**Comment by 1999kevin** [comment]
Hi, do you know if this problem has been solved? I also wonder if it is a ZYZ configuration. If not, what is the configuration? I also try to reconstruct the data while having 2 ratoting axises but te results are not so good

**Comment by AnderBiguri** [comment]
Sorry @pingchanglee , I had completely missed this. Are you still struggling with it? @1999kevin yes its ZYZ as the demos show. What is the issue?

**Reply by 1999kevin** [reply]
Thank you very much. A new issue has been posted.

**Reply by pingchanglee** [reply]
@AnderBiguri 
It is my bad. 
After going over the paper again today, I find that the coordinate system described in the paper is not the same as TIGRE but I carelessly missed it before! 
Now, I have no problem about this topic. Thank you guys!
